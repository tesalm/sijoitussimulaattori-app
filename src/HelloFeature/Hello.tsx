import React from 'react';
import { Button, Platform, Text, View } from 'react-native';
import { connect } from 'react-redux';
import { bindActionCreators, Dispatch } from 'redux';

import { t } from '../assets/i18n';
import { RootState } from '../redux/reducers';
import { addToCounter, removeFromCounter } from './actions';
import CounterView from './components/CounterView';
import { helloStyles as styles } from './styles';

const instructions = Platform.select<string>({
  ios: 'You are on iOS',
  android: 'You are on Android',
});

export interface HelloProps {
  name?: string;
  counterValue: number;
  onIncrement: typeof addToCounter;
  onDecrement: typeof removeFromCounter;
}

interface HelloState {
  amount: number;
}

class Hello extends React.Component<HelloProps, HelloState> {
  constructor(props: HelloProps) {
    super(props);
    this.state = { amount: 1 };
  }

  render() {
    const { name, counterValue, onIncrement, onDecrement } = this.props;

    return (
      <View style={styles.container}>
        <Text style={styles.welcome}>
          {t('Hello.HelloGreetee', { greetee: name || t('Hello.World') })}
        </Text>
        <Text style={styles.instructions}>{instructions}</Text>
        <CounterView counterValue={counterValue} />
        <Button
          onPress={() => onIncrement && onIncrement(this.state.amount)}
          title="Increment counter"
        />
        <Button onPress={onDecrement} title="Decrement counter" />
        <Button
          onPress={() => this.setState({ amount: this.state.amount + 1 })}
          title="Add 1 to incrementation step"
        />
        <Text style={styles.instructions}>
          Incrementation step: {this.state.amount || 1}
        </Text>
      </View>
    );
  }
}

const mapStateToProps = (state: RootState) => ({
  counterValue: state.counter.counterValue,
});

const mapDispatchToProps = (dispatch: Dispatch) =>
  bindActionCreators(
    {
      onIncrement: addToCounter,
      onDecrement: removeFromCounter,
    },
    dispatch
  );

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Hello);

export { Hello as HelloTest };
