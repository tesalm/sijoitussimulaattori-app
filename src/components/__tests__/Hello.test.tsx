import * as React from 'react';
import renderer from 'react-test-renderer';

import { mount } from 'enzyme';

import Hello from '../Hello';

describe('Hello World', () => {
  it('renders correctly', async () => {
    const component = renderer.create(<Hello />).toJSON();
    expect(component).toMatchSnapshot();
  });

  it('has property name correctly set', async () => {
    const wrapper = mount(<Hello name="John" />);
    expect(wrapper.prop('name')).toEqual('John');
  });
});
