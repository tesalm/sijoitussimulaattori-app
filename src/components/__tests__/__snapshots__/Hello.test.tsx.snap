// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hello World renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F5FCFF",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 20,
        "margin": 10,
        "textAlign": "center",
      }
    }
  >
    Hello, 
    World
    !
  </Text>
  <Text
    style={
      Object {
        "color": "#333333",
        "marginBottom": 5,
        "textAlign": "center",
      }
    }
  >
    You are on iOS
  </Text>
</View>
`;
